### DISCUSSION TITLE
onChange > track cell changes

### DISCUSSION THREAD

**Discussion by joanvf** [nick-keller/react-datasheet-grid#141] [Feature requests]
Hi,

Just started using the component after migrating from react-virtualized and it's great, congrats for building this. There is something I'm missing, though, and wanted to share the suggestion. It would be great if the onChange callback gave information on the specific cells that have changed (row+column). Many times we persist cells individually, similar to Google Sheets. Currently we have to compare all row values to identify those that have changed. Since editing is performed at cell level, it could make sense to have callbacks at cell level too, rather than full rows.

Many thanks!

**Comment by nick-keller** [comment]
Hey @joanvf , and thx for your interest!
This is somewhat of a recurring request that I have not figured out yet, but happy to work on it.
How would you imagine the API of this should look like?

Things to consider:
- rows might not be objects, and if it is, cells might not map to one of the object's key
- a cell can impact multiple keys of a row
- multiple cells can impact the same key of a row
- editing a cell directly is just one way to edit its content, you can copy paste, expand selection with corner, duplicate, insert, delete rows...

**Comment by joanvf** [comment]
Hi Nick,

I was thinking something similar to the current onFocus/onBlur callbacks, that provide a cell (or an array of cells if needed).
However, bear in mind we are not currently using the add/delete row functionality, so have not properly thought about how the data model would be if this was used. We would probably need to have some kind of row ID, otherwise the addition/deletion of rows would make the data model very difficult to maintain as row numbers change...
